contracts/Avocado.sol

```diff
@@ -0,0 +1,128 @@
+pragma solidity ^0.4.17;
+
+contract Avocado {
+    enum PersonType { Teacher, Student }
+
+    // Person Structure
+    struct Person {
+        // Teacher or Student
+        PersonType personType;        
+        string name;        
+        string description;                        
+        uint ethPerHour;
+
+        // Array of the Hash of the active meetings
+        bytes32[] activeMeetings;
+
+        // Array of the Hash of completed meetings
+        bytes32[] completedMeetings;
+                
+        bool exists;
+    }
+
+    // Meeting structure
+    struct Meetings {
+        address student;
+        address teacher;
+        string description;
+        uint32 timeInEpoch; // When the session's gonna happen
+        uint8 meetingDuration; // How long is the meeting going to be        
+    }
+
+    // Reviews
+    struct Ratings {
+        address addr;
+        uint rating; // out of 5
+        string description; // Short review description
+    }
+
+    // Hash of Meeting -> IsTeacher (Based on UserType) -> Review Object
+    mapping(bytes32 => mapping(bool => Ratings[])) public ratings;
+
+    // Users using the platform
+    mapping(address => Person) public users;
+
+    // Tags to filter out users
+    // string -> isTeacher (Based on user types) -> Addresses
+    mapping(bytes32 => mapping(bool => address[])) public tags;
```

Comment:
Why use `bytes32` instead of `address`? It's just an ID I assume? Is it because we want to use addresses only for things that relate to Ethereum addresses? If it's just an ID, can we not type alias this so it's more readable?

Reply:
Are we talking about ratings or tags? If it's for ratings I fudged up. If it's for tags it because ethereum doesn't support dynamic length keys

Reply:
@kendricktan I'm still not sure why tags are an array of addresses?

Reply:
Some pseudo-code for what's in my head:

```
struct Tag {...}

alias tagId = bytes32
alias isTeacher = bool

mapping(tagId => mapping(isTeacher => Tag[]))
```